src/components/MapLegend.vue

```diff
@@ -108,24 +126,118 @@
                     legend.appendChild(item);
                 }
             }
+
+            let legendMinus = document.getElementById("minus");
+            let legendPlus = document.getElementById("plus");
+            let legendContent = document.getElementById("legendContent");
+            let legendHeader = document.getElementById("legendHeader");
+
+            let toggle = function(clicked, switchOut){
+                if(clicked.id === "minus"){
+                    legendContent.style.display = "none";
+                }else{
+                    legendContent.style.display = "block";
+                }
+                clicked.style.display = "none";
+                switchOut.style.display = "block";
+            }
+
+            legendMinus.onclick = function(){
+                toggle(legendMinus, legendPlus);
+            }
+            legendPlus.onclick = function(){
+                toggle(legendPlus, legendMinus);
+            }
         },
     }
   }
 </script>
 
 <style scoped lang="scss">
+$border: 1px solid rgb(200, 200, 200);
   .map-overlay {
     font-family: 'merriweather',serif;
-    border-style: groove;
     position: absolute;
     z-index:1000;
     bottom: 15px;
     left: 10px;
     background: rgba(255, 255, 255, 0.8);
     margin-right: 120px;
-    overflow: auto;
+    border: $border;
     border-radius: 3px;
-    padding: 5px;
   }
 
+  #legendHeader{
+      display: flex;
+      border-bottom: $border;
+  }
+
+  #legendTitle{
+      flex: 1;
+
+    p{
+        line-height: 25px;
+        padding: 0 0 0 10px;
+        margin-right: 10px;
+    }      
+  }
+
+  #legendToggle{
+      width:25px;
+      height: 25px;
+      border-left: $border;
+  }
+
+  .legendIcon{
+      display: block;
+      width: 100%;
+      height: 100%;
+      text-align: center;
+      outline: none;
+      cursor: pointer;
+        svg {
```

Comment:
What does this 'svg' section do?

Reply:
Tells the fontawesome icon what size to be and the margin is to center it vertically in the box I made for it.